[WIP] bluetooth functionality

Happy0 opened the pull request.

Happy0 committed:
Start adding bluetooth functionality.

Happy0 committed:
more

Happy0 committed:
Start working towards architecture andre suggested.

staltz committed:
dx: use correct ssb-config version in nodejs-project

staltz committed:
ux: fix color of notification icon to be adaptive

staltz committed:
ux: fix notifications getting stuck (not clearing)

staltz committed:
dx: tweak release commit convention

Happy0 committed:
Getting there... I think.

staltz commented on android/app/build.gradle:
```diff
@@ -164,6 +170,7 @@ android {
 }
 
 dependencies {
+    compile project(':react-native-bluetooth-serial')
```
Just a note: this should be `implementation`, not `compile`

staltz commented on android/gradle.properties:
```diff
@@ -21,3 +21,8 @@ android.useDeprecatedNdk=true
+MYAPP_RELEASE_STORE_FILE=my-release-key.keystore
+MYAPP_RELEASE_STORE_PASSWORD=blahblahblah
+MYAPP_RELEASE_KEY_PASSWORD=blahblahblah
+MYAPP_RELEASE_KEY_ALIAS=my-key-alias
```
Okay for now, but let's definitely not let this get committed and merged :)

Happy0 replied:
haha! yes. it was just the easiest way to get it building completely :P. i plan to tidy up a lot before the 'proper' pull request.

Happy0 committed:
Hooray! it's somewhat alive. my beautiful frankenstein's monster :P

Happy0 committed:
Refactor to make puppet-bluetooth-manager.ts have the same interface as it will have in a linux / mac / windows scuttlebot.

Happy0 committed:
Use bluetooth multiserv plugin.

Happy0 committed:
work with node byte buffers.

Happy0 committed:
Set react native bluetooth serial to github commit.

Happy0 committed:
Don't double encode JSON string.

Happy0 committed:
Fix encoding issues

Happy0 committed:
Fix github commit reference for react-native-serial-bluetooth fork.

Happy0 committed:
Use published version of multiserver-bluetooth.

Happy0 committed:
Moved the scuttlebot bluetooth plugin functionality to a separate module.

staltz committed:
dx: rename mmmmm to manyverse

staltz committed:
ux: remove alpha version disclaimer

Happy0 committed:
Remove unused files (they were moved to separate NPM modules.

staltz committed:
dx: update package-lock.json

staltz committed:
ux: use circular avatars everywhere (component Avatar)

staltz committed:
dx: remove old unused React components

staltz committed:
dx: move DrawerMenuItem into drawer's view

staltz committed:
ux: show self avatar in compose fields

staltz committed:
ux: improve looks of placeholder avatars

staltz committed:
release: 0.0.19-alpha

Happy0 committed:
Use published version of react-native-bluetooth-serial.

Happy0 committed:
Try synced writes.

staltz committed:
dx: target android API 26 (for play store)

Happy0 committed:
Merge remote-tracking branch 'upstream/master' into bluetooth

Happy0 committed:
Woops, add ws config back in.

staltz committed:
dx: update sbot to 12.0.1 in nodejs-project

staltz committed:
ux: pasted invites can also be DHT invites

Happy0 committed:
Use bridge branches of react-native-bluetooth-serial and ssb-mobile-bluetooth-manager.

Happy0 committed:
bump react-native-bluetooth-serial.

staltz committed:
dx: use multiserver 1.13.4 not a fixed commit

staltz committed:
dx: ignore nodejs-assets in prettier

staltz committed:
dx: update ssb-dht-invite to 0.9.0 which actually works

staltz committed:
ux: new Create Invite screen

staltz committed:
dx: tiny refactor in tsconfig.json

staltz committed:
ux: add Share button to Create Invite screen

staltz committed:
dx: refactor AndroidBackButton to be generic

staltz committed:
dx: improve looks of share button in Create Invite screen

staltz committed:
ux: add connectivity mode button in connections tab

staltz committed:
dx: tiny refactor in connections tab view

staltz committed:
ux: fix LAN peer discovery bugs

staltz committed:
ux: connections list shows DHT peers

staltz committed:
ux: show staged (DHT) peers in connections tab

staltz committed:
dx: only periodically ping 1.1.1.1 if connections tab is visible

Happy0 committed:
Helper function for connecting to bluetooth address.

Happy0 committed:
Comment out unused line (but helpful for illustration for now.).

staltz committed:
ux: auto export secrets to upcoming beta version

staltz committed:
dx: add e2e tests for dht invites

staltz committed:
release: 0.0.20-alpha

staltz committed:
dx: rename mmmmm to Manyverse in new files

staltz committed:
dx: activate import of keys from alpha to beta

staltz committed:
ux: improve in-app explanations of dht sync

staltz committed:
dx: update nodejs-project package-lock.json

staltz committed:
dx: update gradle to 3.1.4

staltz committed:
dx: new update-version script uses 0.yy.m.d format

staltz committed:
dx: fix tiny assertion in e2e tests

staltz committed:
release: 0.18.9-21.beta

staltz committed:
dx: improve internal docs

staltz committed:
ux: fix bug report email address

staltz committed:
release: 0.18.9-25.beta

staltz committed:
dx: respect package-lock.json in nodejs-project

staltz committed:
release: 0.18.9-25b.beta

staltz committed:
dx: do not commit android/app/bin

staltz committed:
ux: require Android OS 5.0 or higher

staltz committed:
ux: attempt to fix #128, has-internet false positives

staltz committed:
dx: release apks to two different Dat archives (all, latest)

staltz committed:
dx: update readme with some links

staltz committed:
ux: connections tab update has-internet quicker

staltz committed:
release: 0.18.9-26.beta

staltz committed:
dx: remove versionCode and versionName from manifest

staltz committed:
dx: instruct about sdkmanager for compiling nodejs-mobile

Happy0 committed:
Use release versions of bluetooth libraries.

staltz committed:
dx: update utp-native-no-prebuild

So it compiles more reliably for Android using different kinds of NDK
versions.

staltz committed:
dx: update utp-native-no-prebuild again

Happy0 committed:
;-;

staltz committed:
dx: update nodejs-mobile to 0.3.0

staltz committed:
ux: fix #160, feed scrolling flickering

staltz committed:
dx: use nodejs-mobile's appDataDir instead of our hack

staltz committed:
ux: remove alpha-to-beta implicit migration feature

Happy0 committed:
Add connection types back into the config.

staltz committed:
ux: fix #161, toast to inform profile pics cant be edited

staltz committed:
dx: add changelog file automation

staltz committed:
dx: fix minor typo in readme

staltz committed:
dx: remove leftover debug logger

staltz committed:
ux: improve UI responsiveness and resilience to freezing

Fix issue #134

staltz committed:
ux: fix compose screen glitching your profile picture

Fix issue #174

staltz committed:
ux: Fix following status in profile screens

Fix issue #167

staltz committed:
dx: remove some unnecessary sbot plugins

staltz committed:
dx: remove unused .so binary for android

staltz committed:
ux: fix Edit Profile screen prefilling the name incorrectly

staltz committed:
dx: remove leftover debug logging in the backend

staltz committed:
dx: update e2e tests for edit profile screen

staltz committed:
release: 0.18.9-30.beta

staltz committed:
dx: make minify-nodejs-app more robust

staltz committed:
dx: add warning to minify-nodejs-app when building for fdroid

staltz committed:
dx: revert previous commit

staltz committed:
dx: update changelog with links to stores

staltz committed:
dx: add opencollective backers to readme.md

staltz committed:
dx: avoid broken link when there are no opencollective sponsors

Happy0 committed:
...

Happy0 committed:
Merge in master

staltz committed:
release: 0.18.10-02.beta

Happy0 committed:
stuff

Happy0 committed:
Merge remote-tracking branch 'upstream/master' into bluetooth

Happy0 committed:
Use new verions that use unix sockets to bridge bluetooth connections.

Happy0 committed:
woops.

Happy0 committed:
bump
